### PR TITLE
Feat/hook tests

### PR DESCRIPTION
**Pull request by cubeddu**
This PR adds test cases in the `graph-explorer` package:

- `isErrorResponse` in [packages/graph-explorer/src/connector/utils/isErrorResponse.test.ts](https://github.com/aws/graph-explorer/pull/130/files#diff-d33e75ee43df6c942cdf62382443c8fe617f97e9e078a86d5713a1aa3b165940)
- `useEntities` in [packages/graph-explorer/src/hooks/useEntities.test.ts](https://github.com/aws/graph-explorer/pull/130/files#diff-2d63fa70eda510141d5d9715e7379a158d347fdbc3395038904d03de8c3d3462)
- `useEntitiesCounts` in [packages/graph-explorer/src/hooks/useEntitiesCounts.test.ts](https://github.com/aws/graph-explorer/pull/130/files#diff-4aefedfb45c4ae5e8661dc4530301aea5827622276b2555f20a33022674d459e)
- `useNeighborsOptions` in [packages/graph-explorer/src/hooks/useNeighborsOptions.test.ts](https://github.com/aws/graph-explorer/pull/130/files#diff-80fb9b586808f853b0fd951bccdf0c810a0a51fd8fd18d1867d5b3e6c891ef16)
- `usePrevious` in [packages/graph-explorer/src/hooks/usePrevious.test.ts](https://github.com/aws/graph-explorer/pull/130/files#diff-06d2763afb715430c0f203fa5cd94308eb70e8d6a7325aa228e0c5274514f416)
- `useTextTransform` in [packages/graph-explorer/src/hooks/useTextTransform.test.ts](https://github.com/aws/graph-explorer/pull/130/files#diff-d79ab4fb87599bec9474a7232ed9f7dd4f3d2d693737433f4688c1032b9f28bb)

## Motivation
Adding tests ensures the stability and reliability of the hooks in the `graph-explorer` package. It helps identify any potential bugs or issues before they impact the functionality of the application.

## Changes Made
Test cases have been written for each hook mentioned above. These test cases cover various scenarios and edge cases to validate the correct behavior of the hooks.

## Testing
The added test cases have been executed and pass successfully.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/graph-explorer/src/hooks/useEntities.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+const mockEdge1 = {
+    data: {
+        id: "edge1",
+        type: "FRIEND",
+        source: "node1",
+        target: "node2",
+    }
+} as Edge;
+
+describe('useEntities', () => {
```
Comment (agutierrezgit) [inline packages/graph-explorer/src/hooks/useEntities.test.ts:42]: This is good a starting point for testing `useEntities` hook. However it's worth noting that the current test has limited value because `useEntities` performs additional operations internally, such as removing duplication and adding extra properties to nodes, etc... To enhance the test, I would recommend using the `setEntities` function to set the initial state instead of relying on Recoil. Additionally, consider the extra properties that may be added by the setter when comparing the initial entities with the result.

Here's an example:
```javascript
// ... Rest of your code 

describe("useEntities", () => {
  it("returns expected values with default options", async () => {
    const nodes = [mockNode1, mockNode2];
    const edges = [mockEdge1];

    const { result, waitForNextUpdate } = renderHook(
      () => {
        const [entities, setEntities] = useEntities({ disableFilters: true });
        return { entities, setEntities };
      },
      {
        wrapper: RecoilRoot,
      }
    );

    act(() => {
      result.current.setEntities({ nodes, edges });
    });

    await waitForNextUpdate();

    expect(result.current.entities).toEqual({
      nodes: [mockNode1, mockNode2],
      edges: [mockEdge1],
      // Make sure to consider any extra properties added by the setter
      // e.g., additionalProperties: true,
    });
  });
});
```

File: packages/graph-explorer/src/hooks/useNeighborsOptions.test.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    it('should return neighbors options correctly', () => {
+        // Mock return values for useConfiguration and useTextTransform
+
+        (useConfiguration as jest.Mock).mockReturnValue({
+            getVertexTypeConfig: (type: any) => {
+                return { displayLabel: `Label ${type}` };
+            },
+        });
+        (useTextTransform as jest.Mock).mockReturnValue((str: string) => str.toUpperCase());
```
Comment (agutierrezgit) [inline packages/graph-explorer/src/hooks/useNeighborsOptions.test.ts:36]: Just a suggestion, it would be great to also test `useTransform`hook

**Review by agutierrezgit — COMMENTED** [review]
Thanks for this contribution @cubeddu. It's a good starting point to increase test coverage. 
Do you think it's possible to enhance tests for `useEntities` as per my comment? 
@joywa I have checked and all test are passing but I'd like to wait for @cubeddu to come back for a final check. Thanks

File: packages/graph-explorer/babel.config.js [hunk 3]
```diff
@@ -1,6 +1,17 @@
...
+        targets: {
+          node: 'current',
+        },
+      },
+    ],
+    '@babel/preset-react',
+    '@babel/preset-typescript',
+  ],
+  plugins: [
+    '@babel/plugin-syntax-import-meta',
```
[the remark this answers is not in the document]
Comment (cubeddu) [reply packages/graph-explorer/babel.config.js]: Good call, 

I have been able to successfully remove the package.

File: packages/graph-explorer/src/hooks/useNeighborsOptions.test.ts (hunk 2 above)
Comment (cubeddu) [reply packages/graph-explorer/src/hooks/useNeighborsOptions.test.ts:34]: Certainly, I have created a few tests for useTextTransform.

File: packages/graph-explorer/src/hooks/useEntities.test.ts (hunk 1 above)
Comment (cubeddu) [reply packages/graph-explorer/src/hooks/useEntities.test.ts:12]: I agree with your suggestions and have made the necessary changes. Additionally, I have included a few more cases with this updated version.

**Comment by cubeddu** [comment]
@agutierrezgit Thank you for your feedback. I went ahead and made the necessary updates. I am now ready for review.

**Review by agutierrezgit — COMMENTED** [review]
Awesome @cubeddu thanks. @joywa I have double checked and test are passing. I believe this is great initiative so I'll leave it with you for final approval. Thanks

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,2 @@
-node_modules/
-packages/graph-explorer-proxy-server/node_modules/
-packages/graph-explorer/node_modules/
\ No newline at end of file
+**/node_modules/
+**/coverage/
```
Comment (alexey-temnikov) [inline .gitignore:2]: nit:
```suggestion
**/coverage/

```

File: packages/graph-explorer/.npmrc [hunk 5]
```diff
@@ -1,2 +1,3 @@
 auto-install-peers=true
 engine-strict=true
+public-hoist-pattern[]=*
```
Comment (alexey-temnikov) [inline packages/graph-explorer/.npmrc:3]: nit:
```suggestion
public-hoist-pattern[]=*

```

File: packages/graph-explorer/src/hooks/usePrevious.test.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+        expect(result.current).toBeUndefined();
+
+        rerender({ value: 20 });
+        expect(result.current).toEqual(10);
+
+        rerender({ value: 30 });
+        expect(result.current).toEqual(20);
+    });
+});
```
Comment (alexey-temnikov) [inline packages/graph-explorer/src/hooks/usePrevious.test.ts:22]: nit:
```suggestion
});

```

File: pnpm-lock.yaml [hunk 7]
```diff
@@ -1,43 +1,54 @@
-lockfileVersion: '6.1'
+lockfileVersion: '6.0'
```
Comment (alexey-temnikov) [inline pnpm-lock.yaml:1]: nit: Check why the version is downgraded. Is it increase compatibility or we need to use later version locally?

**Review by alexey-temnikov — APPROVED** [review]

**Review by alexey-temnikov — DISMISSED** [review]

**Review by alexey-temnikov — APPROVED** [review]
